nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -1369,8 +1410,55 @@ where
             .await;
         }
 
+        // Once the state has been recovered, we get rid of the unnecessary forks.
+        Self::prune_old_forks(&mut cryptarchia, &mut leader, relays.storage_adapter()).await;
+
         (cryptarchia, leader)
     }
+
+    /// Remove the in-memory storage of stale blocks from the cryptarchia engine
+    /// and of stale `PoL` notes from the `PoL` machinery.
+    /// Furthermore, it removes the deleted block data from storage as well.
+    async fn prune_old_forks(
+        cryptarchia: &mut Cryptarchia,
+        leader: &mut Leader,
+        storage_adapter: &StorageAdapter<Storage, TxS::Tx, BS::BlobId, RuntimeServiceId>,
+    ) {
+        let old_forks_pruned = cryptarchia.prune_old_forks().collect::<Vec<_>>();
+        // Trigger parallel block deletion and keep a reference to each block ID for
+        // logging purposes (below).
+        let block_deletion_outcomes = join_all(
+            old_forks_pruned
+                .iter()
+                .map(|f| async move { (f.id(), storage_adapter.remove_block(f.id()).await) }),
+        )
+        .await;
+
+        for old_fork_pruned in old_forks_pruned {
+            let block_id = old_fork_pruned.id();
+            if !leader.prune_notes_at(&block_id) {
+                tracing::warn!(
+                    target: LOG_TARGET,
+                    "Leader proof for block {block_id} was not found in storage."
+                );
+            }
```

Comment:
I thikn this might happen if you haven't set any leader note at all

Reply:
Addressed in https://github.com/logos-co/nomos/pull/1261/commits/679f95d758e2493e8563b7608f10f92775d656a8. We debug log only if we delete a note, which makes more sense.